osquery/tables/system/windows/drivers.cpp

```diff
@@ -9,195 +9,224 @@
+      logWarn("Failed to build driver info list.");
+      devIndex++;
+      continue;
+    }
 
-  auto ret = EnumDeviceDrivers(drvBaseAddr, bytesNeeded, &bytesNeeded);
-  driversCount = bytesNeeded / sizeof(drvBaseAddr[0]);
-  if (ret && (driversCount > 0)) {
-    auto driverPath = static_cast<LPSTR>(malloc(MAX_PATH + 1));
-    auto driverName = static_cast<LPSTR>(malloc(MAX_PATH + 1));
-
-    ZeroMemory(driverPath, MAX_PATH + 1);
-    ZeroMemory(driverName, MAX_PATH + 1);
-
-    for (size_t i = 0; i < driversCount; i++) {
-      if (GetDeviceDriverBaseName(drvBaseAddr[i], driverName, MAX_PATH) != 0) {
-        if (GetDeviceDriverFileName(drvBaseAddr[i], driverPath, MAX_PATH) !=
-            0) {
-          // Removing file extension
-          auto fileExtension = strrchr(driverName, '.');
-          *fileExtension = '\0';
-          loadedDrivers[driverName] = driverPath;
-        } else {
-          loadedDrivers[driverName] = "";
-        }
+    if (SetupDiEnumDriverInfo(
+            devInfoSet.get(), &devInfo, SPDIT_CLASSDRIVER, 0, &drvInfo) ==
+        FALSE) {
+      auto err = GetLastError();
+      if (err == ERROR_NO_MORE_ITEMS) {
+        devIndex++;
+        continue;
       } else {
-        TLOG << "GetDeviceDriverFileName failed (" << GetLastError() << ")";
+        logWarn("Failed to enumerate driver info/", err);
+        devIndex++;
+        continue;
+      }
+    }
+    if (SetupDiGetDriverInfoDetail(devInfoSet.get(),
+                                   &devInfo,
+                                   &drvInfo,
+                                   &drvInfoDetail,
+                                   sizeof(SP_DRVINFO_DETAIL_DATA),
+                                   nullptr) == FALSE) {
+      auto err = GetLastError();
+      if (err != ERROR_INSUFFICIENT_BUFFER) {
+        logWarn("Failed to enumerate driver detailed info.", err);
+        devIndex++;
+        continue;
       }
     }
 
-    free(driverPath);
-    free(driverName);
-  } else {
-    TLOG << "EnumDeviceDrivers failed; array size needed is" << bytesNeeded;
-  }
-
-  free(drvBaseAddr);
-}
-
-QueryData genDrivers(QueryContext& context) {
-  DWORD bytesNeeded = 0;
-  DWORD serviceCount = 0;
-  std::map<std::string, std::string> loadedDrivers;
-  QueryData results;
+    Row r;
+    r["device_id"] = devId;
+    r["inf"] = drvInfoDetail.InfFileName;
+    r["provider"] = drvInfo.ProviderName;
+    r["manufacturer"] = drvInfo.MfgName;
+    r["date"] = std::to_string(osquery::filetimeToUnixtime(drvInfo.DriverDate));
+    r["description"] = drvInfo.Description;
+    ULARGE_INTEGER version;
+    version.QuadPart = drvInfo.DriverVersion;
+    r["version"] = std::to_string(HIWORD(version.HighPart)) + "." +
+                   std::to_string(HIWORD(version.LowPart)) + "." +
+                   std::to_string(LOWORD(version.HighPart)) + "." +
+                   std::to_string(LOWORD(version.LowPart));
+
+    const std::map<std::string, DEVPROPKEY> props = {
```

Comment:
Could you place this outside of the method, `kDriverProperties` or a similar name, is our current style.